internal/api/delete_analysistemplate_v1alpha1.go

```diff
@@ -0,0 +1,56 @@
+package api
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"connectrpc.com/connect"
+	"k8s.io/apimachinery/pkg/types"
+
+	svcv1alpha1 "github.com/akuity/kargo/pkg/api/service/v1alpha1"
+)
+
+func (s *server) DeleteAnalysisTemplate(
+	ctx context.Context,
+	req *connect.Request[svcv1alpha1.DeleteAnalysisTemplateRequest],
+) (*connect.Response[svcv1alpha1.DeleteAnalysisTemplateResponse], error) {
+	if !s.cfg.RolloutsIntegrationEnabled {
+		return nil, connect.NewError(
+			connect.CodeUnimplemented,
+			errors.New("Argo Rollouts integration is not enabled"),
+		)
+	}
+
+	project := req.Msg.GetProject()
+	if err := validateFieldNotEmpty("project", project); err != nil {
+		return nil, err
+	}
+
+	name := req.Msg.GetName()
+	if err := validateFieldNotEmpty("name", name); err != nil {
+		return nil, err
+	}
+
+	if err := s.validateProjectExists(ctx, project); err != nil {
+		return nil, err
+	}
+
+	at, err := s.getAnalysisTemplateFn(ctx, s.client, types.NamespacedName{
+		Namespace: project,
+		Name:      name,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if at == nil {
+		err = fmt.Errorf("AnalysisTemplate %q not found in namespace %q", name, project)
+		return nil, connect.NewError(connect.CodeNotFound, err)
+	}
+
+	if err := s.client.Delete(ctx, at); err != nil {
+		return nil, fmt.Errorf("delete AnalysisTemplate: %w", err)
+	}
```

Comment:
Do we have to get before delete? Can we just blindly delete? If it's not found, the error interceptor will correctly interpret the error and the client will still get end up receiving `connect.CodeNotFound`.

Reply:
Well, this matches what my intuition was telling me as well. But I then looked at the other delete commands and noticed they did a get before deleting the object, so I followed their example instead.

Reply:
It's actually done in the right way in other places as well, except for things that have an alias (where we need to look up, although it _could_ be more efficient if only a name is provided). Given this, I addressed it in [c6bf4b8](https://github.com/akuity/kargo/pull/1691/commits/c6bf4b80e0343634b3287699020abe4f42519970)